src/main/scala/rifs/business/controllers/ApplicationController.scala

```diff
@@ -46,8 +47,77 @@ class ApplicationController @Inject()(val cached: Cached, applications: Applicat
     applications.saveSection(id, sectionNumber, request.body).map(_ => NoContent)
   }
 
-  def completeSection(id: ApplicationId, sectionNumber: Int) = Action.async(parse.json[JsObject]) { implicit request =>
-    applications.saveSection(id, sectionNumber, request.body, Some(LocalDateTime.now())).map(_ => NoContent)
+  def deleteSectionItem(id: ApplicationId, sectionNumber: Int, itemNumber: Int) = Action.async { implicit request =>
+    def hasItemNumber(o: JsObject, num: Int) = o \ "itemNumber" match {
+      case JsDefined(JsNumber(n)) if n == num => true
+      case _ => false
+    }
+
+    applications.fetchAppWithSection(id, sectionNumber).flatMap {
+      case Some((app, os)) =>
+        val doc = os.map(_.answers).getOrElse(JsObject(Seq()))
+        val items = doc \ "items" match {
```

Comment:
That'd empty if sectionNumber is not found, so remainingItems further down would always be empty as well.
That'd mean that saveSection further down would create an empty section for that. Do we want that?